src/Akka.Persistence.MongoDb/Journal/MongoDbJournal.cs

```diff
@@ -490,14 +490,18 @@ protected virtual async Task<long> ReplayAllEventsAsync(ReplayAllEvents replay)
             var seqNoFilter = filters.Any() ? builder.And(filters) : null;
 
 
+
+            IFindFluent<JournalEntry, JournalEntry> find;
             // Need to know what the highest seqNo of this query will be
             // and return that as part of the RecoverySuccess message
 
             //If filters is empty, this will deadlock - so when seqNoFilter is null find all
-            var maxSeqNoEntry = seqNoFilter is null? await _journalCollection.Value.Find(_=> true)
-                .SortByDescending(x => x.Ordering)
-                .Limit(1)
-                .SingleOrDefaultAsync(): await _journalCollection.Value.Find(seqNoFilter)
+            if (seqNoFilter is null)
```

Comment:
Can you change line 490 to
```C#
var seqNoFilter = filters.Any() ? builder.And(filters) : FilterDefinition<JournalEntry>.Empty;
```
instead?

Reply:
Or maybe change the whole filter creation block to
```C#
            var builder = Builders<JournalEntry>.Filter;
            var seqNoFilter = builder.Empty;
            if (fromSequenceNr > 0)
                seqNoFilter &= builder.Gt(x => x.Ordering, new BsonTimestamp(fromSequenceNr));
            if (toSequenceNr != long.MaxValue)
                seqNoFilter &= builder.Lte(x => x.Ordering, new BsonTimestamp(toSequenceNr));
```
instead, so you don't have to deal with null?